web-frontend/js/run-request.js

```diff
@@ -32,19 +32,56 @@ function runRequest() {
   });
 }
 
+function stringify(obj) {
```

Comment:
Nice strategy to clean up the API response! This definitely seems like more the client should have to do. Ideally looks like the API response should be standardized a bit better for easier parsing.

Will create another issue to fix that stuff.